sources/Browser/Core/iTermBrowserManager.swift

```diff
@@ -220,6 +221,144 @@ class iTermBrowserManager: NSObject, WKURLSchemeHandler, WKScriptMessageHandler
             forMainFrameOnly: false,
             worlds: [.page, .defaultClient],
             identifier: UserScripts.consoleLog.rawValue))
+
+        // Inject transparent background CSS for see-through effect (like terminal)
+        // Inline the JS to avoid bundle loading issues
+        let transparentBackgroundJS = """
+        (function() {
+            'use strict';
+            console.log('[iTerm2] Transparent background script loaded');
+
+            // Global transparency level (0.0 = opaque, 1.0 = fully transparent)
+            window.iTermTransparencyLevel = 1.0;
+
+            // Make processedElements and applyTransparency global for external control
+            window.processedElements = new Set();  // Use Set instead of WeakSet for clearing
+            let isProcessing = false;
+
+            // Aggressive style injection with continuous monitoring
+            window.applyTransparency = function() {
+                if (isProcessing) return;  // Prevent re-entry
+                isProcessing = true;
+
+                const transparency = window.iTermTransparencyLevel || 1.0;
+                const bgColor = transparency === 1.0 ? 'transparent' : 'rgba(0,0,0,' + (1 - transparency) + ')';
+
+                const style = document.getElementById('iterm2-transparent-background');
+                if (!style) {
+                    const newStyle = document.createElement('style');
+                    newStyle.id = 'iterm2-transparent-background';
+                    (document.head || document.documentElement).appendChild(newStyle);
+                    console.log('[iTerm2] Style tag injected');
+                }
+                // Update CSS with current transparency level
+                const styleElement = document.getElementById('iterm2-transparent-background');
+                if (styleElement) {
+                    styleElement.textContent = `
+                        /* Universal transparent background - opacity controlled by slider */
+                        *, *::before, *::after {
```

Comment:
This will will:
	•	Kill contrast-preserving backgrounds, overlays, etc.
	•	Break sites that rely on background gradients for visibility.
	•	Nuke background images that aren’t “wallpaper” (e.g. buttons, logos, flags).

Something like this would make more sense:
```
    html,
    body,
    main,
    [role="main"],
    article,
    section,
    .content,
    .container,
    .page,
    .layout,
    .wrapper {
        background-color: ${bgColor} !important;
        background-image: none !important;
    }
```